src/components/Task.tsx

```diff
@@ -1,49 +1,55 @@
-import React from "react"
+import React from "react";
 
-// interface Task {
-//   title: string;
-//   completed: boolean;
-//   key: string;
-// }
-// interface TaskProps {
-//   title: string;
-//   completed: boolean;
-//   itemId: string;
-//   completeTask: (task: Task) => void;
-//   removeTask: (key: string) => void;
-// }
+interface Task {
+  title: string;
+  completed: boolean;
+  id: string;
+}
+
+interface TaskProps extends Task {
+  completeTask: (id: string) => void;
+  removeTask: (id: string) => void;
+}
 
-export const Task = ({ completed, completeTask, itemId, removeTask, title}) => {
+const Task = ({
+  completed,
+  completeTask,
+  id,
+  removeTask,
+  title,
+}: TaskProps): JSX.Element => {
   const checkMark = completed
     ? "m8 13 2.165 2.165a1 1 0 0 0 1.521-.126L16 9"
     : "";
 
   return (
-    <li className="todo-listItem" key={itemId} data-item-key={itemId}>
+    <li className="todo-listItem">
       <svg
         className="todo-listImage"
         strokeLinecap="round"
         strokeLinejoin="round"
-        onClick={() => completeTask(itemId)}
+        onClick={() => completeTask(id)}
       >
         <circle cx="12" cy="12" r="11" />
         <path d={checkMark} fill="none" />
       </svg>
       <div className="todo-listItemInner">
         <p
           className={`todo-listText ${completed ? "text-[#d1d5db]" : ""}`}
-          style={{textDecoration: completed ? "line-through" : ""}}
-          onClick={() => completeTask(itemId)}
+          style={{ textDecoration: completed ? "line-through" : "" }}
+          onClick={() => completeTask(id)}
         >
           {title}
         </p>
         <button
           className="hover:bg-blue-400 group todo-listItemRemove"
-          onClick={() => removeTask(itemId)}
+          onClick={() => removeTask(id)}
         >
           Remove
         </button>
       </div>
     </li>
   );
-}
+};
+
+export default Task;
```

Comment:
The reason for making this default export is because of the way TypeScript handles named exports. Naming both the component `const` and type `interface` the same means you cannot use named exports, but it makes it so that you can import both the component and interface elsewhere with one statement and name.

One of the reasons behind this, as named in the commit message, is to help drive design decisions. Using the Task interface through the rest of the app will ensure that our code is shaped properly anywhere a `Task` is used. You should see this play out as you move forward in the commit history.